### PR TITLE
i18n: Deprecate getI18n, dcnpgettext

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to deprecate two methods of the `i18n` module: `getI18n` and `dcnpgettext`. This should be non-impacting to most consumers, as the other functions exposed by the module are sufficient for the majority of usage. The intent here is to limit the maintenance burden in surfacing internal implementation details, avoiding a commitment to maintaining interface parity with the underlying Jed tooling.

**Testing instructions:**

There should be no application impact in these changes.

Verify localization works as expected (site language respected in Gutenberg UI strings).

Verify the deprecated functions are still exposed on the public interface, but log a deprecated message in their usage:

image

### CODE REVIEW

File: packages/i18n/src/test/index.js [hunk 1]
```diff
@@ -34,13 +34,13 @@ const additionalLocaleData = {
 setLocaleData( localeData, 'test_domain' );
 
 describe( 'i18n', () => {
-	describe( 'dcnpgettext()', () => {
-		it( 'absorbs errors', () => {
-			const result = dcnpgettext( 'domain-without-data', undefined, 'Hello' );
+	it( 'absorbs errors', () => {
+		__( 'Hello', 'domain-without-data' );
```
Comment (gziolo) [inline packages/i18n/src/test/index.js]: Should it call `expect( console ).toHaveErrored();` after each function call instead?

**Review by gziolo — APPROVED** [review]
It looks good. It still needs to be rebeased before we proceed.

**Comment by aduth** [comment]
Realizing that my cleverness with `withDeprecation` might backfire with being detected by ESLint rules, as we've had issues before. I might "dumb it down" before merging.

File: packages/i18n/src/test/index.js (hunk 1 above)
Comment (aduth) [reply packages/i18n/src/test/index.js]: >Should it call `expect( console ).toHaveErrored();` after each function call instead?

Good call. Updated in rebased 1f0c639.

**Comment by aduth** [comment]
>Realizing that my cleverness with `withDeprecation` might backfire with being detected by ESLint rules, as we've had issues before. I might "dumb it down" before merging.

Dumbed down in rebased 1f0c639.

**Comment by aduth** [comment]
FYI I'm seeing the deprecation logged with Yoast active. @omarreiss @atimmer Could you test to see whether the suggested alternatives could serve in place of your `getI18n` usage? For what reason are you using it?

**Comment by atimmer** [comment]
@aduth We currently use it to check if our translations have already been loaded by `setLocaleData`. It's not required but was for extra safety, we can drop it I think.

We have an issue on Yoast SEO to track this: https://github.com/Yoast/wordpress-seo/issues/10945.
